packages/toolkit/src/view/knowledge/KnowledgeBaseView.tsx

```diff
@@ -0,0 +1,226 @@
+      selectedNamespace: store.navigationNamespaceAnchor,
+    })),
+  );
+
+  const deleteKnowledgeBase = useDeleteKnowledgeBase();
+  const { refetch: refetchKnowledgeBases } = useGetKnowledgeBases({
+    accessToken,
+    ownerId: selectedNamespace ?? null,
+    enabled: enabledQuery && !!selectedNamespace,
+  });
+
+  const { data: files } = useListKnowledgeBaseFiles({
+    namespaceId: selectedNamespace ?? null,
+    knowledgeBaseId: selectedKnowledgeBase?.kbId ?? "",
+    accessToken,
+    enabled:
+      enabledQuery && Boolean(selectedNamespace) && Boolean(selectedKnowledgeBase),
+  });
+
+  React.useEffect(() => {
+    if (files) {
+      const hasChunks = files.some((file) => file.totalChunks > 0);
+      setIsProcessed(hasChunks);
+    }
+  }, [files]);
+
+  const handleTabChange = (tab: string) => {
+    setActiveTab(tab);
+  };
+
+  const handleTextOptionChange = (option: Nullable<string>) => {
+    setSelectedTextOption(option);
+  };
+
+  const handleKnowledgeBaseSelect = (knowledgeBase: KnowledgeBase) => {
+    setSelectedKnowledgeBase(knowledgeBase);
+    handleTabChange("upload");
+  };
+
+  const handleDeleteKnowledgeBase = (knowledgeBase: KnowledgeBase) => {
+    setKnowledgeBaseToDelete(knowledgeBase);
+    setShowDeleteMessage(true);
+    setPendingDeletions((prev) => [...prev, knowledgeBase.kbId]);
+  };
+
+  const actuallyDeleteKnowledgeBase = async () => {
+    if (!selectedNamespace || !accessToken || !knowledgeBaseToDelete) return;
+
+    try {
+      await deleteKnowledgeBase.mutateAsync({
+        ownerId: selectedNamespace,
+        kbId: knowledgeBaseToDelete.kbId,
+        accessToken,
+      });
+      setShowDeleteMessage(false);
+      setKnowledgeBaseToDelete(null);
+      setPendingDeletions((prev) =>
+        prev.filter((id) => id !== knowledgeBaseToDelete.kbId),
+      );
+      if (selectedKnowledgeBase?.kbId === knowledgeBaseToDelete.kbId) {
+        setSelectedKnowledgeBase(null);
+        setActiveTab("knowledge-base");
+      }
+      refetchKnowledgeBases();
+    } catch (error) {
+      console.error("Error deleting knowledge base:", error);
+      setShowDeleteMessage(false);
+    }
+  };
+
+  const undoDelete = () => {
+    setShowDeleteMessage(false);
+    if (knowledgeBaseToDelete) {
+      setPendingDeletions((prev) =>
+        prev.filter((id) => id !== knowledgeBaseToDelete.kbId),
+      );
+    }
+    setKnowledgeBaseToDelete(null);
+  };
+
+  const handleProcessFile = () => {
+    setActiveTab("catalog");
+    setShowCreditUsage(true);
+    setTimeout(() => {
+      setShowCreditUsage(false);
+    }, CREDIT_TIMEOUT);
+  };
+
+  const handleGoToUpload = () => {
+    handleTabChange("upload");
+  };
+
+  React.useEffect(() => {
+    let timer: NodeJS.Timeout;
+    if (showDeleteMessage) {
+      timer = setTimeout(() => {
+        actuallyDeleteKnowledgeBase();
+      }, DELETE_KNOWLEDGE_BASE_TIMEOUT);
+    }
+    return () => clearTimeout(timer);
+  }, [showDeleteMessage]);
```

Comment:
@thewbuk The delete mechanism is confusing here, please leave comment to describe it